From ddf475386db93a8317ae507fc841bd689cf60376 Mon Sep 17 00:00:00 2001
From: Daniel Templeton <templedf@apache.org>
Date: Tue, 11 Sep 2018 10:57:11 -0700
Subject: [PATCH 2839/2848] CDH-71491. Revert "Fix potential FSImage
 corruption. Contributed by Ekanth Sethuramalingam
 & Arpit Agarwal."

This reverts commit 81668072d156f23fb267c1728b5ba27bac8320ed.

==C5_APPROVED_BUGFIX==

Change-Id: Id193a51537d9706c2f36a216d9e135324275ae46
---
 .../hdfs/server/namenode/AclEntryStatusFormat.java |    6 ++----
 .../server/namenode/INodeWithAdditionalFields.java |    4 ++--
 2 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/AclEntryStatusFormat.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/AclEntryStatusFormat.java
index 2c5b23b..82aa214 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/AclEntryStatusFormat.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/AclEntryStatusFormat.java
@@ -38,8 +38,7 @@
  * [1:3) -- the type of the entry (AclEntryType) <br>
  * [3:6) -- the permission of the entry (FsAction) <br>
  * [6:7) -- A flag to indicate whether Named entry or not <br>
- * [7:8) -- Reserved <br>
- * [8:32) -- the name of the entry, which is an ID that points to a <br>
+ * [7:32) -- the name of the entry, which is an ID that points to a <br>
  * string in the StringTableSection. <br>
  */
 public enum AclEntryStatusFormat {
@@ -48,8 +47,7 @@
   TYPE(SCOPE.BITS, 2),
   PERMISSION(TYPE.BITS, 3),
   NAMED_ENTRY_CHECK(PERMISSION.BITS, 1),
-  RESERVED(NAMED_ENTRY_CHECK.BITS, 1),
-  NAME(RESERVED.BITS, 24);
+  NAME(NAMED_ENTRY_CHECK.BITS, 25);
 
   private final LongBitFormat BITS;
 
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INodeWithAdditionalFields.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INodeWithAdditionalFields.java
index 4422fbe..c33aaa4f 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INodeWithAdditionalFields.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INodeWithAdditionalFields.java
@@ -35,8 +35,8 @@
     implements LinkedElement {
   static enum PermissionStatusFormat {
     MODE(null, 16),
-    GROUP(MODE.BITS, 24),
-    USER(GROUP.BITS, 24);
+    GROUP(MODE.BITS, 25),
+    USER(GROUP.BITS, 23);
 
     final LongBitFormat BITS;
 
-- 
1.7.9.5

